### PR TITLE
volunteer signup email notification

### PR DESCRIPTION
**Pull request by willblake01**
Send a notification to Mallory whenever there is a new volunteer registration form submission. Allows admin to communicate with volunteers and organize and align volunteer resources in a timely manner. Reduces time and manual work for the volunteering process.

-Setup controller to send email on successful volunteer creation
-Add volunteer notification mailer
-Add volunteer notification mailer views
-Add volunteer notification mailer to rails preview
-Configure test environment to allow for mailers
-Add logo to Event spec factory

[finishes #175443431]

### CODE REVIEW

**Comment by willblake01** [comment]
@ColinW520 Should I move the mailers to methods and call those from the create action instead? There will be at least one more that goes to the volunteer to confirm their form submission.

image

File: app/controllers/volunteers_controller.rb [hunk 1]
```diff
@@ -9,7 +9,8 @@ def create
     @volunteer = Volunteer.new(volunteer_params)
     result = CreateVolunteer.new(volunteer: @volunteer).call
     if result.success
-      redirect_to(root_path, success: result.message)
+      VolunteerRegistrationMailer.with(volunteer: @volunteer).notify.deliver_now
```
Comment (willblake01) [inline app/controllers/volunteers_controller.rb:12]: @ColinW520 Should I move the mailer call to methods instead? There will be at least one more going to the volunteer to confirm their form submission.

File: app/views/mailers/volunteer_notification_mailer/notify.html.haml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+          %tbody
+            %tr
+              %td{:style=>"padding-left: 4px"} #{@volunteer.event.name}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.volunteer_position.name}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.first_name}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.last_name}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.email}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.time}
+              %td{:style=>"padding-left: 4px"} #{@volunteer.created_at.localtime.strftime("%c")}
```
Comment (willblake01) [inline app/views/mailers/volunteer_notification_mailer/notify.html.haml:42]: @ColinW520 I'm repeating myself quite a bit in the inline css. Should I move this to mail.css or should that be reserved for styling that applies to every email?

**Comment by willblake01** [comment]
image

image

image

File: app/controllers/volunteers_controller.rb (hunk 1 above)
Comment (ColinW520) [reply app/controllers/volunteers_controller.rb:12]: This should be done in the service

File: app/views/layouts/volunteer_notification.html.haml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+!!!
+%html{:xmlns => "http://www.w3.org/1999/xhtml"}
+  %head
+    / If you delete this tag, the sky will fall on your head
```
Comment (ColinW520) [inline app/views/layouts/volunteer_notification.html.haml:4]: Sometimes, some mail browsers do things we don't expect. Suggest removing any/all comments.

File: app/mailers/volunteer_notification_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class VolunteerNotificationMailer < ApplicationMailer
+  layout 'volunteer_notification'
```
Comment (ColinW520) [inline app/mailers/volunteer_notification_mailer.rb:2]: Why the specific layout?

Comment (ColinW520) [reply app/mailers/volunteer_notification_mailer.rb:2]: Ah nvm I see

File: app/views/mailers/volunteer_notification_mailer/notify.html.haml (hunk 2 above)
Comment (ColinW520) [reply app/views/mailers/volunteer_notification_mailer/notify.html.haml:42]: nah, mailer styles are super silly. stick with the inline, or, define it in a `<style>` block above, unless `mailer.css` gets injected & reliably outputs what you need.

File: lib/mailer_previews/volunteer_notification/mailer_preview.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class VolunteerNotification::MailerPreview < ActionMailer::Preview
+  def notify
+    VolunteerNotificationMailer.with(volunteer: Volunteer.all.sample).notify
```
Comment (ColinW520) [inline lib/mailer_previews/volunteer_notification/mailer_preview.rb:3]: 💯 for writing a preview!

**Review by ColinW520 — APPROVED** [review]
🚀 

A couple suggestions!

File: app/controllers/volunteers_controller.rb (hunk 1 above)
Comment (willblake01) [reply app/controllers/volunteers_controller.rb:12]: Ok, that's what I thought we discussed but I got an error the first time I tried and the Rails docs had it in the controller, but I guess they're just showing the most basic use case. I have it working from the service now. Thanks!

**Pull request merged**